core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -88,6 +88,20 @@ class LogTest extends JUnitSuite {
       assertEquals("Changing time beyond rollMs and appending should create a new segment.", numSegments, log.numberOfSegments)
     }
 
+    time.sleep(log.config.segmentMs + 1)
+    val setWithTimestamp =
+      TestUtils.singleMessageSet(payload = "test".getBytes, timestamp = time.milliseconds + log.config.segmentMs + 1)
+    log.append(setWithTimestamp)
+    assertEquals("The number of segments should stay the same", 5, log.numberOfSegments)
```

Comment:
Hmm, is that right? The rolling check happens before the append. So, the first message in the segment is "set", which has NoTimestamp. Then, we will be using the creation time of the segment. So, it seems that in the case, we should roll another segment after calling append() in line 94.